client.go

```diff
@@ -120,9 +132,22 @@ func readExpr() <-chan expr {
 		s := bufio.NewScanner(c)
 		for s.Scan() {
 			log.Printf("recv: %s", s.Text())
-			p := newParser(strings.NewReader(s.Text()))
-			if p.parse() {
-				ch <- p.expr
+
+			// recv has to be handled outside of the main thread, which is
+			// blocked when running a synchronous shell command ("$" or "!")
```

Comment:
Thanks for adding this! I'd add one more line of explanation, e.g.

> This is important since `query` is often the result of a user running `$lf -remote "query $id <something>"`.

Reply:
I think there's some issue in GitHub right now which is causing pull requests to not update properly, hopefully it will show up later.